src/test/java/org/folio/rest/api/CheckInStorageApiTest.java

```diff
@@ -0,0 +1,138 @@
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.support.ApiTests;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.JsonResponse;
+import org.folio.rest.support.MultipleRecords;
+import org.folio.rest.support.builders.CheckInBuilder;
+import org.folio.rest.support.http.AssertingRecordClient;
+import org.folio.rest.support.http.InterfaceUrls;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonObject;
+
+public class CheckInStorageApiTest extends ApiTests {
+  private static final String TABLE_NAME = "check_in";
+
+  private final AssertingRecordClient checkInClient =
+    new AssertingRecordClient(client, StorageTestSuite.TENANT_ID,
+      InterfaceUrls::checkInsStorageUrl, "checkIns");
+
+  @Before
+  public void beforeEach() {
+    StorageTestSuite.cleanUpTable(TABLE_NAME);
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs(TABLE_NAME);
+  }
+
+  @Test
+  public void canCreateCheckIn() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = createSampleCheckIn().create();
+
+    IndividualResource createResult = checkInClient.create(checkInToCreate);
+
+    assertThat(createResult.getJson(), is(checkInToCreate));
+  }
+
+
+  @Test
+  public void cannotCreateCheckInIfRequiredPropertyMissing() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = new CheckInBuilder().create();
+
+    JsonResponse createResponse = checkInClient.attemptCreate(checkInToCreate);
+
+    assertThat(createResponse, isValidationResponseWhich(hasMessage("may not be null")));
+    assertThat(createResponse, isValidationResponseWhich(hasParameter("occurredDateTime", "null")));
+  }
+
+  @Test
+  public void canSearchForCheckInsByOccurredDateTime()
+    throws InterruptedException, ExecutionException, TimeoutException,
+    MalformedURLException {
+
+    final DateTime beforeCreateDateTime = DateTime.now(DateTimeZone.UTC);
+
+    JsonObject firstCheckIn = createSampleCheckIn().create();
+    JsonObject secondCheckIn = createSampleCheckIn().create();
+    JsonObject thirdCheckIn = createSampleCheckIn().create();
+
+    checkInClient.create(firstCheckIn);
+    checkInClient.create(secondCheckIn);
+    checkInClient.create(thirdCheckIn);
+
+    MultipleRecords<JsonObject> allCheckInOperations = checkInClient
+      .getMany(String.format("occurredDateTime >= %s", beforeCreateDateTime.getMillis()));
+
+    assertThat(allCheckInOperations.getTotalRecords(), is(3));
+    assertTrue(allCheckInOperations.getRecords().contains(firstCheckIn));
+    assertTrue(allCheckInOperations.getRecords().contains(secondCheckIn));
+    assertTrue(allCheckInOperations.getRecords().contains(thirdCheckIn));
```

Comment:
Given this test is intended to demonstrate searching by when a check in occurred, could it demonstrate how they are filtered by only including some of the records in the expected response?

Reply:
@bohdan-suprun Why was this comment resolved? The test appears to still expect all of the records be returned, which to me, does not demonstrate any filtering by the CQL query.

Reply:
I've added a separate test to verify this case, see `canFilterCheckInsByServicePointAndItem` below.

Reply:
In which case, what is this test intended to demonstrate?

To me, a search that returns all of the records in a system does not demonstrate that the search did any filtering, what am I missing?

Reply:
I've merged those two tests.